packages/cli/src/commands/publish/submit.ts

```diff
@@ -42,19 +43,38 @@ export const publishSubmitCommand = async ({
     release: releaseDetails,
     solana_mobile_dapp_publisher_portal: solanaMobileDappPublisherPortalDetails,
   } = await getConfigFile();
+
   const sign = ((buf: Buffer) =>
     nacl.sign(buf, signer.secretKey)) as SignWithPublisherKeypair;
 
-  await publishSubmit(
-    { connection, sign },
-    {
-      appMintAddress: appMintAddress ?? appDetails.address,
-      releaseMintAddress: releaseMintAddress ?? releaseDetails.address,
-      publisherDetails,
-      solanaMobileDappPublisherPortalDetails,
-      compliesWithSolanaDappStorePolicies,
-      requestorIsAuthorized,
-    },
-    dryRun
-  );
+  const pubAddr = publisherDetails.address;
+  const appAddr = appMintAddress ?? appDetails.address;
+  const releaseAddr = releaseMintAddress ?? releaseDetails.address;
+
+  try {
+    const results = await connection.getMultipleAccountsInfo([
+      new PublicKey(pubAddr),
+      new PublicKey(appAddr),
+      new PublicKey(releaseAddr),
+    ]);
+
+    if (results?.length == 3) {
```

Comment:
NIT: I'd discourage this pattern:
```
if (success_condition) {
  // Do the happy path thing
} else {
  throw new Error("...");
}
```

in favor of:
```
if (!success_condition) {
  throw new Error("...");
}

// Do the happy path thing
```

Rationale: the former nests the success case, which increases the cognitive investment required when reading this code to locate which is the flow that is normal vs. exceptional.

Reply:
Totally correct. I'll clean up this logic and make it a function for the other steps